lib/cooked_post_processor.rb

```diff
@@ -76,8 +77,6 @@ def post_process_images
       limit_size!(img)
       convert_to_link!(img)
     end
-
-    update_post_image
```

Comment:
The idea is that `update_post_image` must be called after `post_process_images` and after oneboxing is applied (`post_process_oneboxes`) to be able to extract `image_url`  for oneboxed images. It doesn't need to be inside the mutex, it can be placed also at the end of the `post_process_oneboxes` method. However, I think this way it is more cleaner how post is being postprocessed.